lib/jekyll/filters/url_filters.rb

```diff
@@ -33,6 +33,21 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces a path relative to the current page. For example, if the path
+      # `/images/me.jpg` is supplied, then the result could be, `me.jpg`,
+      # `images/me.jpg` or `../../images/me.jpg` depending on whether the current
+      # page lives in `/images`, `/` or `/sub/dir/` respectively.
+      #
+      # input - a path relative to the project root
+      #
+      # Returns the supplied path relativized to the current page.
+      def relativize_url(input)
+        return if input.nil?
+        page_url = @context.registers[:page]["url"]
+        page_dir = Pathname(page_url).parent
+        Pathname(input).relative_path_from(page_dir).to_s
```

Comment:
Are there any error scenarios we should be worried about here? What if the file doesn't exist?

We should also ensure the `input` is inside of the `destination` directory or else this could be used for arbitrary path traversal which would be no fun.

Reply:
I don't think we should worry about whether the file exists or not. In the end `{{ "/non/existent.file" | relativize_url }}` is the same as just writing `non/existent.file` directly in markup.

As it happens, `Pathname.relative_path_from` ensures that the result doesn't go outside the root. Added tests for it.